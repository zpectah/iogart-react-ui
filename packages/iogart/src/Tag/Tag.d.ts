import { TagProps } from './types';

declare const Tag: TagProps;

export default Tag;

