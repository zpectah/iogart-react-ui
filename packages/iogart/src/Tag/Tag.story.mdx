import { Meta, Story } from '@storybook/addon-docs';

import Tag from './Tag';
export const Template = (args) => <Tag {...args} />;

<Meta
    title="Components/Status/Tag"
    component={Tag}
/>

# Button
`@iogart-react-ui/iogart/Tag`

```javascript
import { Tag } from 'iogart-react-ui/iogart';
```

<Story
  name="Default"
  args={{
    label: 'Tag name',
  }}>
  {Template.bind({})}
</Story>
