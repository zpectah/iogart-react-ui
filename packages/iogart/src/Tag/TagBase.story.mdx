import { Meta, Story } from '@storybook/addon-docs';

import TagBase from './TagBase';
export const Template = (args) => <TagBase {...args} />;

<Meta title="Components/Status/Tag" component={TagBase} />

# tag Base

`@iogart-react-ui/base/TagBase`

```javascript
import { TagBase } from 'iogart-react-ui/base';
```

<Story
  name="Base"
  args={{
    label: 'Tag Base',
  }}
>
  {Template.bind({})}
</Story>
