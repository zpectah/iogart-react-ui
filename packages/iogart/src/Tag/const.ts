export const TAG_BASE_CLASSNAME = 'TagBase';
export const TAG_CLASSNAME = 'Tag';
