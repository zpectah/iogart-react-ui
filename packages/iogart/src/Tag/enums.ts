export enum TagElementKeys {
    span = 'span',
    div = 'div',
}
