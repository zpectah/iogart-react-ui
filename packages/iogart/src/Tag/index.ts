export { default as useTagBase } from './useTagBase';
export { default as useTag } from './useTag';
export { default as TagBase } from './TagBase';
export { default as Tag } from './Tag';
export * from './const';
export * from './types';
