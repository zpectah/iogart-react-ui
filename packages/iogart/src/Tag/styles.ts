import { createUseStyles } from 'react-jss';

const useTagStyles = createUseStyles({
  root: {},
});

export default useTagStyles;
