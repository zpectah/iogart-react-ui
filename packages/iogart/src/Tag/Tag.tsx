import React, { createElement, useMemo } from 'react';

import { IogartButtonElement } from '@iogart-react-ui/types';
import { useButtonBase } from '@iogart-react-ui/base';
import { Close } from '@iogart-react-ui/icons';
import { TagElementKeys } from './enums';
import { TagProps } from './types';
import useTag from './useTag';

const Tag = (props: TagProps) => {
    const {
        elementType = TagElementKeys['span'],
        label = '',
        clickable,
        deleteIcon,
        onClickDelete,
        ...rest
    } = props;

    const { ...restOfUpdatedProps } = useTag({ clickable, ...rest });

    const buttonProps = useButtonBase({
        className: 'iogart-tag-button',
        onClick: onClickDelete && onClickDelete,
    }) as IogartButtonElement;

    const childrenNode = useMemo(() => {
        let children = label;
        if (clickable && onClickDelete) {
            const closeButton = (
                <button {...buttonProps}>
                    {deleteIcon ? deleteIcon : <Close />}
                </button>
            );
            children = (
                <>
                    {children}
                    {closeButton}
                </>
            );
        }

        return children;
    }, [ label, clickable, onClickDelete, deleteIcon, buttonProps ]);

    return createElement(
        elementType,
        { ...restOfUpdatedProps },
        childrenNode,
    );
};

export default Tag;
