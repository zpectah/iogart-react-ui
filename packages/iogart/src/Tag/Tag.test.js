import renderer from 'react-test-renderer';

import Tag from './Tag';

it('Tag - onMouseEnter & onMouseLeave', () => {

    const component = renderer.create(
        <Tag>Tag</Tag>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    renderer.act(() => {
        tree.props?.onMouseEnter();
    });

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    renderer.act(() => {
       tree.props?.onMouseLeave();
    });

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

});
