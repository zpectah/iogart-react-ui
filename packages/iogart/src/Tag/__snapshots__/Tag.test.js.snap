// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag - onMouseEnter & onMouseLeave 1`] = `
<span
  className="iogart-ui iogart-tag"
  onClick={[Function]}
  onDoubleClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={{}}
/>
`;

exports[`Tag - onMouseEnter & onMouseLeave 2`] = `
<span
  className="iogart-ui iogart-tag iogart--hover"
  onClick={[Function]}
  onDoubleClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={{}}
/>
`;

exports[`Tag - onMouseEnter & onMouseLeave 3`] = `
<span
  className="iogart-ui iogart-tag"
  onClick={[Function]}
  onDoubleClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={{}}
/>
`;
